sharedbuffers/mapped_struct.py

```diff
@@ -320,6 +320,342 @@ def unpack_from(cls, buf, offs, idmap = None, array = array.array, itemsizes = {
             raise ValueError("Inconsistent data, unknown type code %r" % (dcode,))
         return rv
 
+class mapped_dict(dict):
+
+    @classmethod
+    def pack_into(cls, obj, buf, offs, idmap = None, implicit_offs = 0):
+        return mapped_list.pack_into((obj.keys(), [obj[k] for k in obj]), buf, offs, idmap, implicit_offs)
```

Comment:
I'd rather build both lists explicitly. Why? Because mapped_object is very pedant about the type of things. If .keys() returns something that's a sequence but not really a list, this will fail.

So, I'd suggest:

``` python
keys = list(obj)
values = [ obj[k] for k in keys ]
return mapped_list.pack_into([keys, values], buf, offs, idmap, implicit_offs)
```